src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1891,6 +1911,65 @@ impl CheckValve {
     }
 }
 
+pub struct PriorityValve {
+    open_ratio: LowPassFilter<Ratio>,
+
+    fully_closed_threshold: Pressure,
+    fully_opened_threshold: Pressure,
+
+    upstream_pressure: Pressure,
+    downstream_pressure: Pressure,
+}
+impl PriorityValve {
+    const VALVE_RESPONSE_TIME_CONSTANT: Duration = Duration::from_millis(5);
+
+    pub fn new(fully_closed_threshold: Pressure, fully_opened_threshold: Pressure) -> Self {
+        Self {
+            open_ratio: LowPassFilter::<Ratio>::new(Self::VALVE_RESPONSE_TIME_CONSTANT),
+
+            fully_closed_threshold,
+            fully_opened_threshold,
+
+            upstream_pressure: Pressure::default(),
+            downstream_pressure: Pressure::default(),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext, upstream_pressure: Pressure) {
+        self.upstream_pressure = upstream_pressure;
+
+        self.update_open_state(context);
+
+        self.update_downstream_pressure();
+    }
+
+    fn update_open_state(&mut self, context: &UpdateContext) {
+        let opening_ratio = if self.upstream_pressure > self.fully_opened_threshold {
+            Ratio::new::<ratio>(1.)
```

Comment:
Do we need to cover this case separately? I feel like this is covered by the `.min` below.